addons/account/models/account_move.py

```diff
@@ -130,6 +130,25 @@ def _get_default_invoice_incoterm(self):
         ''' Get the default incoterm for invoice. '''
         return self.env.company.incoterm_id
 
+    @api.model
+    def _company_terms(self, company):
+        company_terms_note = company.terms_type != 'html'
+        if company_terms_note:
+            return company.invoice_terms if not is_html_empty(company.invoice_terms) else ''
+        baseurl = html_keep_url(self.env.company.get_base_url() + '/terms')
+        return _('Terms & Conditions: %s', baseurl)
+
+    @api.model
+    def _default_narration(self):
+        use_invoice_terms = self.env['ir.config_parameter'].sudo().get_param('account.use_invoice_terms')
+        if not use_invoice_terms:
+            return ''
+        narration = self._company_terms(self.company_id)
+        if not narration:
+            narration = self._company_terms(self.env.company)
+        return narration
```

Comment:
```suggestion
        return self._company_terms(self.company_id) or self._company_terms(self.env.company)
```
Could be simplified to this but I'm not sure it's even worth it given the simplicity of the code.